hybrid/pkg/server/server.go

```
@@ -0,0 +1,297 @@
+				Request: &nodeattestorv1.AttestRequest_Payload{
+					Payload: val,
+				},
+			}
+
+			newInterceptor := NewServerInterceptor()
+			newInterceptor.SetReq(newReq)
+			newInterceptor.setCustomStream(stream)
+			interceptors[index] = newInterceptor
+
+			elem := reflect.ValueOf(plugin.Plugin)
+			result := elem.MethodByName("Attest").Call([]reflect.Value{reflect.ValueOf(newInterceptor)})
+
+			if result[0].Interface() != nil {
+				callError, _ := status.FromError(result[0].Interface().(error))
+				return status.Errorf(codes.Internal, callError.Message())
+			}
+		} else {
+			return status.Errorf(codes.InvalidArgument, "plugin %v not found", plugin.PluginName)
+		}
+	}
+
+	return p.SendResponse(interceptors, stream)
+}
+
+func (p *HybridPluginServer) SendResponse(interceptors []ServerInterceptor, stream nodeattestorv1.NodeAttestor_AttestServer) error {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	selectors := []string{}
+	canReattest := true
+	spiffeId := interceptors[0].SpiffeID()
+
+	for _, interceptor := range interceptors {
+		if !p.checkReattestation(interceptor.CanReattest()) {
+			attested, err := p.CheckAttested(interceptor.Context(), interceptor.SpiffeID())
+			canReattest = false
+			switch {
+			case err != nil:
+				return err
+			case attested:
+				return status.Error(codes.PermissionDenied, "attestation data has already been used to attest an agent")
+			default:
+			}
+		}
+
+		selectors = append(selectors, interceptor.CombinedSelectors()...)
+	}
+
+	return stream.Send(&nodeattestorv1.AttestResponse{
+		Response: &nodeattestorv1.AttestResponse_AgentAttributes{
+			AgentAttributes: &nodeattestorv1.AgentAttributes{
+				CanReattest:    canReattest,
+				SpiffeId:       spiffeId,
+				SelectorValues: selectors,
+			},
+		},
+	})
+}
+
+func (p *HybridPluginServer) Configure(ctx context.Context, req *configv1.ConfigureRequest) (*configv1.ConfigureResponse, error) {
+	pluginData, _ := p.decodeStringAndTransformToAstNode(req.HclConfiguration)
```

error is ignored, this must not be done, to avoid unexpected errors

---

done :)